pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/TransformFunctionFactory.java

```diff
@@ -198,6 +199,9 @@ public static TransformFunction get(ExpressionContext expression, Map<String, Da
         FunctionContext function = expression.getFunction();
         String functionName = canonicalize(function.getFunctionName());
         List<ExpressionContext> arguments = function.getArguments();
+        if (functionName.equalsIgnoreCase("AS")) {
+          arguments = arguments.subList(0, 1);
+        }
```

Comment:
why not just pass both of them in? it shouldn't matter as long as `AsTransformFunction` handles it correctly yes?

Reply:
cause the other side is an identifier but not defined in schema, it will cause underly data block initialization error with NPE.